src/server.js

```diff
@@ -44,7 +45,11 @@ export default function hypernova(userConfig, onServer) {
 
   logger.init(config.logger, config.loggerInstance);
 
-  const app = createApplication();
+  if (typeof config.createApplication !== 'function') {
+    throw new TypeError('Hypernova requires a `createApplication` property and it must be a function that return a valid express instance');
+  }
+
+  const app = config.createApplication();
```

Comment:
can we also validate that `app` is an express instance?

Reply:
That's a bit of a tricky one, `express().constructor.name` gives us `EventEmitter` so the only way we see how to do this would be to check for all the functions we expect to we available, the ones used in `worker.js`. So we could do something like:
```js
if (
  typeof this.express.use !== 'function' &&
  typeof this.express.post !== 'function' &&
  typeof this.express.listen !== 'function'
) {
  throw new TypeError(
    '`createApplication` must return a valid express instance with `use`, `post` and `listen` methods'
  );
}
```

Reply:
Fair enough, that seems reasonable :-)